contrib/scalar/scalar.c

```diff
@@ -984,6 +984,73 @@ void load_builtin_commands(const char *prefix, struct cmdnames *cmds)
 	die("not implemented");
 }
 
+static void dir_file_stats(struct strbuf *buf, const char *path)
+{
+	DIR *dir = opendir(path);
+	int dir_fd = dirfd(dir->dirp);
+	struct dirent *e;
+	struct stat e_stat;
+
+	if (!dir)
+		return;
+
+	strbuf_addstr(buf, "Contents of ");
+	strbuf_add_absolute_path(buf, path);
+	strbuf_addstr(buf, ":\n");
+
+	while ((e = readdir(dir)) != NULL)
+		if (!is_dot_or_dotdot(e->d_name) &&
+		    !fstatat(dir_fd, e->d_name, &e_stat, 0))
+			strbuf_addf(buf, "%-70s %16lld\n", e->d_name,
+			            e_stat.st_size);
+}
+
+static int count_files(char *path)
+{
+	DIR *dir = opendir(path);
+	struct dirent *e;
+	int count = 0;
+
+	if (!dir)
+		return 0;
+
+	while ((e = readdir(dir)) != NULL)
+		if (!is_dot_or_dotdot(e->d_name) && e->d_type == DT_REG)
+			count++;
+
+	return count;
+}
+
+#define IS_HEX_DIGIT(c) (((c) - '0') <= '9' - '0' || \
+                         ((c) - 'A') <= 'F' - 'A' || \
+                         ((c) - 'a') <= 'f' - 'a')
```

Comment:
Is this assuming an `unsigned char`?

```suggestion
#define IS_HEX_DIGIT(c) (('0' <= (c) && (c) <= '9') || \
                         ('A' <= (c) && (c) <= 'F') || \
                         ('a' <= (c) && (c) <= 'f'))
```

Also, it seems you have spaces instead of tabs here. You can fix with `git rebase --whitespace=fix`.